### PR TITLE
[CI] Fix dependency installation

### PR DESCRIPTION
**Pull request by phschiele**
## Description
Starting by simplification of dependency installation. Many cases are meanwhile obsolete.
Issue link (if applicable): CI failing, e.g. in #1840 #1867 #1872 

## Type of change
- [ ] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [x] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [ ] Add our license to new files.
- [ ] Check that your code adheres to our coding style.
- [ ] Write unittests.
- [ ] Run the unittests and check that they’re passing.
- [ ] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

File: continuous_integration/install_dependencies.sh [hunk 1]
```diff
@@ -10,48 +10,25 @@ conda config --set remote_max_retries 10
 conda config --set remote_backoff_factor 2
 conda config --set remote_read_timeout_secs 120.0
 
-if [[ "$PYTHON_VERSION" == "3.6" ]]; then
-  conda install scipy=1.3 numpy=1.16 mkl pip=21.3.1 pytest pytest-cov lapack ecos scs osqp cvxopt
-  python -m pip install cplex sdpa-python  # CPLEX is not available yet on 3.10
-elif [[ "$PYTHON_VERSION" == "3.7" ]] || [[ "$PYTHON_VERSION" == "3.8" ]]; then
+if [[ "$PYTHON_VERSION" == "3.7" ]] || [[ "$PYTHON_VERSION" == "3.8" ]]; then
   conda install scipy=1.3 numpy=1.16 mkl pip pytest pytest-cov lapack ecos scs osqp cvxopt
-  python -m pip install cplex sdpa-python  # CPLEX is not available yet on 3.10
```
Comment (phschiele) [inline continuous_integration/install_dependencies.sh:18]: CPLEX is available on 3.10 now.

File: continuous_integration/install_dependencies.sh [hunk 2]
```diff
@@ -10,48 +10,25 @@ conda config --set remote_max_retries 10
 conda config --set remote_backoff_factor 2
 conda config --set remote_read_timeout_secs 120.0
 
-if [[ "$PYTHON_VERSION" == "3.6" ]]; then
```
Comment (phschiele) [inline continuous_integration/install_dependencies.sh:13]: Remove all 3.6 cases.

File: continuous_integration/install_dependencies.sh [hunk 3]
```diff
@@ -10,48 +10,25 @@ conda config --set remote_max_retries 10
...
 if [[ "$PYTHON_VERSION" == "3.9" ]] || [[ "$RUNNER_OS" == "Windows" ]]; then
   conda install pyscipopt"<4.0"  # TODO: update interface https://github.com/cvxpy/cvxpy/pull/1628
 fi
 
-if [[ "$PYTHON_VERSION" == "3.10" ]]; then
-  python -m pip install diffcp gurobipy
-elif [[ "$PYTHON_VERSION" == "3.6" ]]; then
-  python -m pip install diffcp xpress
-else
-  python -m pip install diffcp gurobipy xpress
```
Comment (phschiele) [inline continuous_integration/install_dependencies.sh:52]: All now available on 3.7-3.10.

**Comment by phschiele** [comment]
Note:       
During build, even though numpy 1.16.6 is already installed, it is installed again, which subsequently fails.
```
Collecting numpy<1.17,>=1.16
Downloading numpy-1.16.6.zip (5.1 MB)
```

**Comment by phschiele** [comment]
@SteveDiamond As you can see in the [actions tab](https://github.com/cvxpy/cvxpy/actions/runs/2870945262), this PR fixes the CI issues. It fails again in the latest commit because I added back `ortools`, which are fixed separately in #1872.

-> Tests should pass again after merging both PRs.

**Comment by SteveDiamond** [comment]
Great, thanks @phschiele !

**Pull request merged**